include/drivers/syscon.h

```diff
@@ -0,0 +1,70 @@
+/*
```

Comment:
I think its better this header live in drivers/syscon/

Reply:
I don't think this can live in `drivers/syscon` because this header file should be included by any other driver that needs to access the syscon area.

Reply:
We just need to add `zephyr_library_include_directories` in CMakeLists.txt for it.

Reply:
how is this better? As a developer I expect the exposed driver interface to be somewhere in `include/drivers/` while in `drivers/xxx` I expect to find the driver-specific header files.